thumbor_aws/loader.py

```diff
@@ -57,69 +58,84 @@
 )
 
 
-async def load(context, path):
-    """Loader to get source files from S3"""
-    client = S3Client(context)
-    client.configuration = {
-        "region_name": context.config.AWS_LOADER_REGION_NAME,
-        "secret_access_key": context.config.AWS_LOADER_S3_SECRET_ACCESS_KEY,
-        "access_key_id": context.config.AWS_LOADER_S3_ACCESS_KEY_ID,
-        "endpoint_url": context.config.AWS_LOADER_S3_ENDPOINT_URL,
-        "bucket_name": context.config.AWS_LOADER_BUCKET_NAME,
-        "root_path": context.config.AWS_LOADER_ROOT_PATH,
-    }
-    if client.compatibility_mode is True:
-        client.configuration["region_name"] = context.config.TC_AWS_REGION
-        client.configuration["endpoint_url"] = context.config.TC_AWS_ENDPOINT
-        client.configuration[
-            "bucket_name"
-        ] = context.config.TC_AWS_LOADER_BUCKET
-        client.configuration[
-            "root_path"
-        ] = context.config.TC_AWS_LOADER_ROOT_PATH
-
-    bucket, real_path = get_bucket_and_path(
-        client.configuration["bucket_name"], path
-    )
-
-    norm_path = normalize_url(client.configuration["root_path"], real_path)
-    result = LoaderResult()
-
-    status_code, body, last_modified = await client.get_data(
-        bucket, norm_path, expiration=None
-    )
-
-    if status_code != 200:
-        result.error = LoaderResult.ERROR_NOT_FOUND
-        result.extra = body
-        result.successful = False
-        return result
-
-    result.successful = True
-    result.buffer = body
+class S3Loader:
```

Comment:
I get why you added a class to make it a singleton but we can make a singleton just as easily with methods:

```python
__instance = None

def load(params):
    if __instance is None:
        __instance = Whatever()

    # do something with __instance
```